app/src/org/commcare/tasks/SendTask.java

```diff
@@ -91,6 +94,26 @@ protected Boolean doTaskBackground(Void... params) {
                 continue;
             }
             try {
+
+                // see if we have a form.properties file to load the PostURL from
+                FilenameFilter filter = new FilenameFilter() {
+                    @Override
+                    public boolean accept(File dir, String filename) {
+                        return filename.equals(ZipTask.FORM_PROPERTIES_FILE);
+                    }
+                };
+
+                File[] fileContents = f.listFiles(filter);
+                if(fileContents != null && fileContents.length > 0){
+                    Properties properties = FileUtil.loadProperties(fileContents[0]);
+                    if(properties != null && properties.getProperty(ZipTask.FORM_PROPERTY_POST_URL) != null){
+                        url = properties.getProperty(ZipTask.FORM_PROPERTY_POST_URL);
+                        Logger.log(TAG, "Successfully got form.property PostURL: " + url);
+                    }
+                    // don't submit this file
+                    FileUtil.deleteFileOrDir(fileContents[0]);
+                }

```

Comment:
I thought about it but I think this is cleaner because 1) we can only override url if all the requisite condition are met and 2) we'll probably add more properties at some point and I didn't want to a map back and forth